src/templates.rs

```diff
@@ -52,6 +60,47 @@ impl Templates {
         }
     }
 
+    #[allow(clippy::too_many_arguments)]
+    pub fn get_body_by_implicit_name(
+        &self,
+        organization_id: Option<&str>,
+        environment_name: Option<&str>,
+        starts_with: Option<&str>,
+        ends_with: Option<&str>,
+        contains: Option<&str>,
+        export: bool,
+        secrets: bool,
+        template_name: &str,
```

Comment:
I like that you're not leaking `get_implicit_template_query::ImplicitTemplateFilters` here as the argument type, but it might be helpful to declare a new struct named `TemplateFilters` and then a conversion trait to `get_implicit_template_query::ImplicitTemplateFilters`. See [From](https://doc.rust-lang.org/std/convert/trait.From.html) and [Into](https://doc.rust-lang.org/std/convert/trait.Into.html). That'd allow better grouping of the arguments and avoid such a long list.

Reply:
This is a great suggestion, and I spent some trying to implement it but I'm having some difficulty and going to time box for now and push up the changes from your other comments.